test/src/test/java/org/corfudb/infrastructure/log/StreamLogCompactionTest.java

```diff
@@ -31,24 +44,28 @@ public void testCompaction() throws InterruptedException {
         StreamLog streamLog = mock(StreamLog.class);
         doThrow(new RuntimeException("err")).when(streamLog).compact();
 
-        StreamLogCompaction compaction = new StreamLogCompaction(
-                streamLog, initialDelay, period, TimeUnit.MILLISECONDS, Duration.ofSeconds(1)
-        );
+        final long initialCompactionCounter = getCompactionCounter();
+        StreamLogCompaction compaction = new StreamLogCompaction(streamLog,
+                                                                 initialDelay,
+                                                                 period,
+                                                                 TimeUnit.MILLISECONDS,
+                                                                 PARAMETERS.TIMEOUT_VERY_SHORT);
 
-        final long expectedGcCounter = 2;
-        while(getGcCounter() < expectedGcCounter){
+        final long expectedCompactionCounter = 2 + initialCompactionCounter;
+        while(getCompactionCounter() < expectedCompactionCounter){
             TimeUnit.MILLISECONDS.sleep(timeout);
         }
 
         compaction.shutdown();
 
-        assertThat(getGcCounter()).isGreaterThanOrEqualTo(expectedGcCounter);
+        assertThat(getCompactionCounter()).isGreaterThanOrEqualTo(expectedCompactionCounter);
     }
 
-    private long getGcCounter() {
-        return ServerContext.getMetrics()
-                    .getCounters()
-                    .get(StreamLogCompaction.STREAM_COMPACT_METRIC)
-                    .getCount();
+    private long getCompactionCounter() {
+        final Timer compactionTimer = metricRegistry.timer(StreamLogCompaction.STREAM_COMPACT_METRIC);
+
+        return compactionTimer == null ?
```

Comment:
As I know, if timer doesn't exists it will be created, so no need to check for null

Reply:
Done.